### PR TITLE
Feature/v1-219 | Implement scheduler for updating test statuses.

### PR DESCRIPTION
**Pull request by dapodev**
# Overview

Implemented scheduler that starts it's job every day at 7 AM and updates statuses of client courses from `failed` to `started` if `disable test time` expired.

# Changes
1. Implemented scheduler using `node-cron`.
2. Refactored db connection.

### CODE REVIEW

File: backend/src/utils/schedule/constants.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { ISchedulerTimeOptions } from 'interfaces/common/scheduler';
+
+export const UPDATE_TEST_STATUSES_JOB_START_HOURS = 7;
+
+export const SCHEDULER_OPTIONS: Record<string, ISchedulerTimeOptions> = {
+  UPDATE_TEST_STATUSES: {
+    hour: String(UPDATE_TEST_STATUSES_JOB_START_HOURS),
+    minute: '0',
+    second: '0',
+  } as const,
```
Comment (SaXaPhonist) [inline backend/src/utils/schedule/constants.ts:10]: why do you need here 2 "as const"

**Review by SaXaPhonist — CHANGES_REQUESTED** [review]

File: backend/src/utils/date/testDate.ts [hunk 2]
```diff
@@ -1,5 +1,8 @@
 import moment from 'moment';
 
+import { DISABLE_TEST_DAYS } from 'config/constants';
+import { UPDATE_TEST_STATUSES_JOB_START_HOURS } from 'utils/schedule/constants';
+
 /**
```
[the remark this answers is not in the document]
Comment (dapodev) [reply backend/src/utils/date/testDate.ts:6]: I'll remove it

File: backend/src/middlewares/connectionMiddleware.ts [hunk 3]
```diff
@@ -1,23 +1,11 @@
 import { NextFunction, Request, Response } from 'express';
-import { connect } from 'mongoose';
 
 import ServiceUnavailableError from 'classes/errors/serverErrors/ServiceUnavailableError';
+import { connectToDatabase } from 'utils/connection';
 
 const connectionMiddleware = async (req: Request, res: Response, next: NextFunction) => {
   try {
-    const DB_URL: Record<string, string | undefined> = {
```
[the remark this answers is not in the document]
Comment (dapodev) [reply backend/src/middlewares/connectionMiddleware.ts:8]: I don't think we should copy/paste this code among the project. I will leave it that way

File: backend/src/utils/schedule/constants.ts (hunk 1 above)
Comment (dapodev) [reply backend/src/utils/schedule/constants.ts]: removed

**Review by SaXaPhonist — APPROVED** [review]

**Pull request merged**